playwright-junit/src/test/java/com/microsoft/playwright/junit/BrowserCustomFactoryTests.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.playwright.junit;
+
+import com.microsoft.playwright.Browser;
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@UseBrowserFactory(CustomBrowserFactory.class)
+public class BrowserCustomFactoryTests {
+  private Browser browserFromBeforeEach;
+  private static Browser browserFromBeforeAll;
+
+  @BeforeAll
+  static void beforeAll(Browser browser) {
+    assert browser != null;
+    System.out.println("BeforeAll " + browser);
+    browserFromBeforeAll = browser;
+    assertEquals("firefox", browser.browserType().name());
+  }
+
+  @AfterAll
+  static void afterAll(Browser browser) {
+    assert browser != null;
+    System.out.println("AfterAll " + browser);
+    assertEquals(browserFromBeforeAll, browser, "Static Browser is not equal to others that were created");
+  }
+
+  @BeforeEach
+  void beforeEach(Browser browser) {
+    assert browser != null;
```

Comment:
`assertNotNull(browser);` here and in other places so that we use junit assertions in the tests and not rely on java assertions to be enabled?

Reply:
I was thinking of removing these tests altogether and using playwright-junit in the actual playwright tests.  what do you think?  that way we won't need a separate test suite for this.  If we do want a separate test suite, JUnit provides a way to test extensions that we could use to properly test the extensions.

Reply:
That'd be nice and would also help us evaluate usability of the fixtures.

Reply:
@yury-s I'm running into a circular dependency issue when trying to update the playwright tests to using playwright-junit.

Since playwright-junit needs playwright as a dependency, do you know of a clean way to use playwright-junit in playwright?  

i can think of 2 ways:

1.  include junit-playwright inside of the main playwright package but with the junit dependency as provided scope.  
2. put junit-playwright in a completely separate maven project

What are your thoughts?

Reply:
If I understand correctly, marking `playwright` dependency of `playwright-junit` `provided` should enable using it in the `playwright` tests, is it not working? I believe you need to remove `playwright` from the parent pom.xml in that case.

Reply:
If this does not work, including junit stuff into the main playwright module with "provided" junit dependency is okay too.

Reply:
Yea i don't think it will work.  If i remove playwright from the parent pom, then maven complains it can't find it inside `playwright-junit`.  And once i add `playwright-junit` to `playwright`, i get

```
java: Annotation processing is not supported for module cycles. Please ensure that all modules from cycle [playwright,playwright-junit] are excluded from annotation processing
```

Reply:
I'll just include it in the main playwright module